### PR TITLE
fix(translation) translation button visibility for users, add missing…

### PR DESCRIPTION
**Pull request by tamasdomokos**
… languages

### CODE REVIEW

File: react/features/base/i18n/i18next.ts [hunk 1]
```diff
@@ -35,22 +37,18 @@ const COUNTRIES = _.merge({}, COUNTRIES_RESOURCES, COUNTRIES_RESOURCES_OVERRIDES
...
-/**
- * The languages for the top section of the translation language list.
- *
- * @public
- * @type {Array<string>}
- */
-export const TRANSLATION_LANGUAGES_HEAD: Array<string> = [ 'en' ];
+export const i18n = {
+    get translationLanguages() {
+        const config: IConfig = APP.store.getState()['features/base/config'];
```
Comment (saghul) [inline react/features/base/i18n/i18next.ts:42]: The APP global does not exist on mobile. What's your plan there?

Comment (tamasdomokos) [reply react/features/base/i18n/i18next.ts:42]: I could just use the defaults if's not defined(but then we can't overwrite the languages). I'm not sure what how could i do this on mobile. But if you can help with this i'm open to it.

**Comment by damencho** [comment]
Why do we need `lang/translation-languages-enGB.json` and `lang/translation-languages.json` ?

**Comment by tamasdomokos** [comment]
> Why do we need `lang/translation-languages-enGB.json` and `lang/translation-languages.json` ?

I was following the the existing logic. But yes, i was just duplicating the same file. I can remove.

File: config.js [hunk 2]
```diff
@@ -374,6 +374,12 @@ var config = {
     // transcription: {
     //     // Whether the feature should be enabled or not.
     //     enabled: false,
+    
+    //     //Available languages for the language selector dialog.
+    //     translationLanguages: ["en", "es", "fr", "ro" ...]
```
Comment (saghul) [inline config.js:379]: Make the comment actual working JS if uncommented please.

File: config.js [hunk 3]
```diff
@@ -374,6 +374,12 @@ var config = {
...
+
+    //     //Important languages to show on the top of the language list.
+    //     translationLanguagesHead: ["en", ...]
```
Comment (saghul) [inline config.js:382]: Ditto. Also add the trailing comma

File: react/features/base/config/configWhitelist.ts [hunk 4]
```diff
@@ -178,6 +178,7 @@ export default [
     'hideEmailInSettings',
     'hideLobbyButton',
     'hosts',
+    'i18n',
```
Comment (saghul) [inline react/features/base/config/configWhitelist.ts:181]: This is no longer necessary is it?

File: react/features/subtitles/components/LanguageSelectorDialog.web.tsx [hunk 5]
```diff
@@ -6,35 +6,37 @@ import { IState } from '../../app/types';
...
 // @ts-ignore
-import { updateTranslationLanguage, setRequestingSubtitles, toggleLangugeSelectorDialog } from '../actions';
+import { updateTranslationLanguage, setRequestingSubtitles, toggleLanguageSelectorDialog } from '../actions';
 
 import LanguageList from './LanguageList';
 
 interface ILanguageSelectorDialogProps {
     _language: string;
     t: Function;
+    translationLanguages: Array<string>;
```
Comment (saghul) [inline react/features/subtitles/components/LanguageSelectorDialog.web.tsx:19]: style: props that come from maptStateToProps should begin with `_`

File: react/features/subtitles/components/LanguageSelectorDialog.web.tsx [hunk 6]
```diff
@@ -6,35 +6,37 @@ import { IState } from '../../app/types';
...
 
-    const importantLanguages = TRANSLATION_LANGUAGES_HEAD.map((lang: string) => `languages:${lang}`);
-    const fixedItems = [ off, ...importantLanguages ];
-
-    const languages = LANGUAGES
-        .filter((lang: string) => !TRANSLATION_LANGUAGES_EXCLUDE.includes(lang))
-        .map((lang: string) => `languages:${lang}`)
-        .filter((lang: string) => !(lang === language || importantLanguages.includes(lang)));
+    const languagesHead = translationLanguagesHead.map((lang: string) => `translation-languages:${lang}`);
+    const fixedItems = [ off, ...languagesHead ];
```
Comment (saghul) [inline react/features/subtitles/components/LanguageSelectorDialog.web.tsx:36]: What does "fixed" mean here? I feel like a comment explaining what you're doing here would help.

File: react/features/subtitles/components/LanguageSelectorDialog.web.tsx [hunk 7]
```diff
@@ -81,17 +83,18 @@ const LanguageSelectorDialog = ({ _language }: ILanguageSelectorDialogProps) =>
...
-        conference
-    } = state['features/base/conference'];
+    const { conference } = state['features/base/conference'];
+    const { _language } = state['features/subtitles'];
+    const { transcription } = state['features/base/config'];
 
-    const {
-        _language
-    } = state['features/subtitles'];
+    const languages = transcription?.translationLanguages || TRANSLATION_LANGUAGES;
```
Comment (saghul) [inline react/features/subtitles/components/LanguageSelectorDialog.web.tsx:90]: use `??` for default values.

File: react/features/subtitles/middleware.js [hunk 8]
```diff
@@ -115,7 +116,7 @@ function _endpointMessageReceived({ dispatch, getState }, next, action) {
                 newTranscriptMessage));
 
         } else if (json.type === JSON_TYPE_TRANSCRIPTION_RESULT
-                && !translationLanguage) {
+        && i18next.language === translationLanguage) {
```
Comment (saghul) [inline react/features/subtitles/middleware.js:119]: Why is this check performed?

Comment (tamasdomokos) [reply react/features/subtitles/middleware.js:119]: There is an edge case when a user is selecting the same language as the transcription. In this case both source language and destination language is **en** and the google translation api returning an error that the two languages are equal. By doing this check the user can display the transcription, and i don't have the remove the source language for the list of available languages and transcription is also available.

File: config.js (hunk 2 above)
Comment (tamasdomokos) [reply config.js]: done

File: config.js (hunk 3 above)
Comment (tamasdomokos) [reply config.js]: done

File: react/features/base/config/configWhitelist.ts (hunk 4 above)
Comment (tamasdomokos) [reply react/features/base/config/configWhitelist.ts]: removed

File: react/features/subtitles/components/LanguageSelectorDialog.web.tsx (hunk 7 above)
Comment (tamasdomokos) [reply react/features/subtitles/components/LanguageSelectorDialog.web.tsx]: done

File: config.js [hunk 9]
```diff
@@ -1349,6 +1357,8 @@ var config = {
      */
     mouseMoveCallbackInterval: 1000,
 
+    hiddenDomain: 'tdomokos.jitsi.net',
```
Comment (saghul) [inline config.js:1360]: debug leftover?

**Review by saghul — APPROVED** [review]

**Pull request merged**